scripts/build/termux_step_override_config_scripts.sh

```diff
@@ -3,11 +3,28 @@ termux_step_override_config_scripts() {
 		return
 	fi
 
+	# resolves many variants of "cannot execute binary file: Exec format error"
+	# symlink host binaries over incompatible architecture and libc crossbuild rootfs binaries
+	symlink_incompatible_binary() {
+		src="$1"
+		dst="$2"
+		if [ -f $src ] && ! file $(readlink -f $dst) | grep text >/dev/null \
+		   && [ "$(readlink -f $src)" != "$(readlink -f $dst)" ]; then
+			echo "symlink_incompatible_binary: linking $dst to $src"
+			ln -sf $src $dst
+		fi
+	}
+	export -f symlink_incompatible_binary
+	find $TERMUX_PREFIX/bin/ -executable -exec bash -c 'symlink_incompatible_binary /usr/bin/$(basename {}) {}' \;
+	unset symlink_incompatible_binary
+	# from one perspective, the above block could be considered an expansion of the same logic
+	# behind the line below this, but for everything in the usr/bin folder instead of only bin/sh
+
 	# Make $TERMUX_PREFIX/bin/sh executable on the builder, so that build
 	# scripts can assume that it works on both builder and host later on:
 	ln -sf /bin/sh "$TERMUX_PREFIX/bin/sh"
 
-	if [ "$TERMUX_INSTALL_DEPS" = false ] || [ "$TERMUX_PACKAGE_LIBRARY" = "glibc" ]; then
+	if [ "$TERMUX_PACKAGE_LIBRARY" = "glibc" ]; then
```

Comment:
Please move this logic to the `[ "$TERMUX_ON_DEVICE_BUILD" = true ]` logic which is at the beginning of the function.
```bash
if [ "$TERMUX_ON_DEVICE_BUILD" = true ] || [ "$TERMUX_PACKAGE_LIBRARY" = "glibc" ]; then
	return
fi
```

Reply:
Oh ok, so to clarify that as much as possible, you are saying that the prexisting line `ln -sf /bin/sh "$TERMUX_PREFIX/bin/sh"` should _not_ be executed if `TERMUX_PACKAGE_LIBRARY` is `glibc`? 

That would be a removing preexisting code from the glibc codepath, which I have not tested. Would you like me to test building all glibc packages with this PR as well?

Reply:
> Oh ok, so to clarify that as much as possible, you are saying that the preexisting line ln -sf /bin/sh "$TERMUX_PREFIX/bin/sh" should not be executed if TERMUX_PACKAGE_LIBRARY is glibc?

Yes and no. Your change now replaces not only `sh`, but all possible binaries/scripts from Termux packages. Because of this, I have a concern that if the build script does not skip this stage when compiling for glibc packages, it may break the build of some packages. To understand: in some glibc packages, binaries or scripts from glibc packages may be used (i.e. runned) for compilation.

Reply:
> all possible binaries/scripts from Termux packages

This part ensures that _only_ files that are _not_ identified as "text" by `file` command are modified

but I understand, if the glibc packages need to use the `aarch64` binaries directly during their builds, it needs to be like you suggested. I applied that change in my recent push.